### ISSUE TITLE
Use git-repo

### ISSUE THREAD

**Issue by jayvdb** [meetmangukiya/cobot#2]
[git-repo](https://github.com/guyzmo/git-repo) has a fairly neat set of functionality that would allow all of the commands to work with 'any' git hosting service.

It provides an abstraction layer on top of specialised libraries for each git hosting service, and has a good (but basic) test suite, so we wouldnt need to worry about how to contact these services, even if they are private repos.

**Comment by nemani** [comment]
You meant to open this here? Or https://github.com/meetmangukiya/err-you-cobot/? 
I don't think we are working in this version of cobot.

**Comment by jayvdb** [comment]
No, if we're going to be using `err`, I think we would want a new plugin in a new & separate repo, as many other people would be wanting to use it.

If we're building a new bot from scratch (this repo), then I think we'd want to make it git-centric (but multi hosting service), and putting git-repo as one of the pillars of the bot would help do that.

**Comment by meetmangukiya** [comment]
If this is something like an API wrapper that is more easier to use then it is awesome. There is always an alternative that we can call the command needed for the task using `subprocess` and get the job done. But that looks little dirty. 

Unified interface to interact with various git-hosting website is surely interesting though.
